google/cloud/storage/internal/rest/stub.cc

```diff
@@ -106,6 +106,36 @@ StatusOr<ReturnType> CreateFromJson(
   return ReturnType::CreateFromJson(*payload);
 }
 
+Status AddCustomHeaders(Options const& options,
+  RestRequestBuilder& builder) {
+    // In tests this option may not be set. And over time we want to retire it.
+    if (!options.has<CustomHeadersOption>()) return {};
+    auto custom_headers =
+        options.get<CustomHeadersOption>();
+    for (auto const& h : options.get<CustomHeadersOption>()) {
+      builder.AddHeader(h.first, h.second);
+    }
+    return {};
```

Comment:
I'm surprised this doesn't cause an error as the function is declared returning a `Status`. I would check what this actually does when it returns.